lib/src/system_cache.dart

```diff
@@ -100,8 +98,8 @@ class SystemCache {
   }
 
   /// Returns the version of [source] bound to this cache.
-  BoundSource source(Source source) =>
-      _boundSources.putIfAbsent(source, () => source.bind(this));
+  BoundSource source(Source? source) =>
```

Comment:
I don't think the input should be nullable:
```suggestion
  BoundSource source(Source source) =>
```